### PR TITLE
[Standby] Filter by categories in the  cataloge page

### PR DESCRIPTION
**Pull request by sallemiines**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description   |This PR allow to : Filter by categories in the catalog page
| Type?         |  new feature
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | /
| How to test?  | Run the script    => path=full/02_product/16_filter_by_categories_in_cataloge_page npm run specific-test -- --URL=BackOfficeURL

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9296)

### CODE REVIEW

File: tests/E2E/test/clients/category.js [hunk 1]
```diff
@@ -4,47 +4,58 @@ global.checkCategoryName = [];
...
 class Category extends CommonClient {
 
-    checkImage(selector) {
+  checkImage(selector) {
+    return this.client
+      .waitForExist(selector)
+      .pause(2000)
+      .then(() => this.client.isExisting(selector))
+      .then((text) => expect(text).to.be.equal(true));
+
```
Comment (fatmaBouchekoua) [inline tests/E2E/test/clients/category.js]: Please remove empty line

File: tests/E2E/test/selectors/BO/add_product_page.js [hunk 2]
```diff
@@ -224,6 +229,16 @@ module.exports = {
...
     page_active_number: '//*[@id="product_catalog_list"]//li[@class="page-item active"]//input',
-    item_per_page: '//*[@id="paginator_select_page_limit"]'
+    item_per_page: '//*[@id="paginator_select_page_limit"]',
+    category_list: '//*[@id="product_catalog_list"]/div[2]/div/table/tbody/tr[%ID]/td[5]',
+    category: '//*[@id="choice_tree"]/ul/li/ul/li[%ID]/div/label',
+    subCat: '//*[@id="choice_tree"]/ul/li/ul/li[%I]/ul/li[%J]/div/label',
+    subCat_number: '//*[@id="choice_tree"]/ul/li/ul/li[%ID]/ul/li',
+    categories_filters: '//*[@id="product_catalog_list"]//table/tbody/tr[%ID]/td[6]',
+    product_name_link: '//*[@id="product_catalog_list"]//table/tbody/tr[%ID]/td[4]/a',
+    pencil: '//*[@id="product_catalog_list"]/div[2]/div/table/tbody/tr[%ID]/td[10]/div/div/a[1]/i',
```
Comment (fatmaBouchekoua) [inline tests/E2E/test/selectors/BO/add_product_page.js]: This selector is not used please remove it

**Review by fatmaBouchekoua — COMMENTED** [review]
Almost good except some little changes

**Review by fatmaBouchekoua — APPROVED** [review]

**Comment by marionf** [comment]
Hello @sallemiines 
After filtering by the "accessories" category & checking results, we need to click again on "Filter by categories", unselect and check again results like in the video: https://drive.google.com/file/d/1c3fyJiHDy3_TooWq6EiL5OmP0IsLZ4VR/view

**Pull request closed without merging**